src/Components/Assets/AssetType/ONVIFCamera.tsx

```diff
@@ -71,34 +71,50 @@ const ONVIFCamera = (props: ONVIFCameraProps) => {
 
   const handleSubmit = async (e: SyntheticEvent) => {
     e.preventDefault();
-    if (checkIfValidIP(cameraAddress)) {
-      setLoadingSetConfiguration(true);
-      setIpAddress_error("");
-      const data = {
-        meta: {
-          asset_type: "CAMERA",
-          middleware_hostname: middlewareHostname, // TODO: remove this infavour of facility.middleware_address
-          local_ip_address: cameraAddress,
-          camera_access_key: `${username}:${password}:${streamUuid}`,
-        },
-      };
-      const res: any = await Promise.resolve(
-        dispatch(partialUpdateAsset(assetId, data))
-      );
+    if (
+      middlewareHostname.trim() === "" ||
+      cameraAddress.trim() === "" ||
+      username.trim() === "" ||
+      password.trim() === "" ||
+      streamUuid.trim() === ""
+    ) {
+      Notification.Error({
```

Comment:
We do not want such error notifications for missing fields. Instead, maintain a state which will store all field related errors. Then, Just after `e.preventDefault();` here, call a `validateForm` method which will loop through all the fields we have and check for their validity and if they are invalid (missing), then update the error state with the field name with the error, `errors[field] = "Please select the xxx field";`. Have this error state linked to all the `TextFormField`  as `<TextFormField ..... error={state.errors['field_name']}`. So we will have a red text box below the input with the error.

Take reference from [UserAdd.tsx](https://github.com/coronasafe/care_fe//blob/develop/src/Components/Users/UserAdd.tsx#L367)

Required error message design (sample):
![image](https://github.com/coronasafe/care_fe/assets/3626859/cb673c99-30ad-4cde-be39-f664d408fe70)


Current error message design on this PR:
![image](https://github.com/coronasafe/care_fe/assets/3626859/6bff4d8e-46d5-4cd7-978a-883091c66328)
![image](https://github.com/coronasafe/care_fe/assets/3626859/7242d14e-a7ec-4db7-9ea7-7bfa214a6d7c)

Suggestion by @nihal467 . Feel free to ask any clarifications on this.